exasol/toolbox/tools/dependencies_check.py

```diff
@@ -0,0 +1,35 @@
+from tomlkit import loads
+from sys import exit
+import pytest
+
+
+FILTER = ['url', 'git', 'path']
+
+
+def _source_filter(version, filter):
+    output = None
+    if isinstance(version, dict):
+        for key in version.keys():
+            if key in filter:
+                output = key
+    return output
+
+
+def dependency_check():
+    with open("pyproject.toml", 'r') as toml:
+        toml = loads(toml.read())
+    dependencies = []
+    for name, version in toml["tool"]["poetry"]["dependencies"].items():
```

Comment:
1. Please make sure, the implementation does not fail, if one of these sections is missing.
E.g. a file `pyproject.toml` without section `[tool.poetry.dependencies]`.
2. Please include other dependencies, too, e.g. `[tool.poetry.dev-dependencies]`

Reply:
You can do so by 
* either catching exception `KeyError`, 
* using [dict.get()](https://docs.python.org/3/library/stdtypes.html#dict.get), 
* or checking `if key in dict`